src/tool.rs

```diff
@@ -16,6 +16,8 @@ use crate::{
     Error, ErrorKind, OutputKind,
 };
 
+pub(crate) type CompilerFamilyLookupCache = HashMap<(Box<Path>, Vec<String>), ToolFamily>;
```

Comment:
I prefer to store program and args together in one box to reduce size of key and avoid repeated allocation:

```suggestion
pub(crate) type CompilerFamilyLookupCache = HashMap<Box<[OsStr]>, ToolFamily>;
```

Reply:
thought: That suggestion does lose some type-level guarantees about the arguments being `String`s. I don't see a concrete problem with it, but I'll give it a shot!

Implemented with latest force push. Interdiff link: https://github.com/rust-lang/cc-rs/compare/c5c1a91ad6532e70081ead71ffc81485df116932..ffe0e4164b4d6ac31c62aed7888651d3d15f0c4f